### PR TITLE
Don't modulate canvas on invisible color change

### LINKED ISSUE #10098: Setting color on a hidden CanvasModulate node modulates all open scenes
**Operating system or device - Godot version:**
Windows 7, both Godot 2.1.3 and 3.x have this problem

**Steps to reproduce:**
Have multiple 2D scenes open(not mandatory), create a CanvasModulate node, tick visibility off, change the color.

**Expected result:**
Hidden CanvasModulate has no effect

**Actual result:**
All open scenes are modulated with the chosen color, unless they have their own, non-hidden CanvasModulate node.

EDIT: Should have clarified, this only has effect inside of editor.

### PR DESCRIPTION
**Pull request by KoBeWi**
Fixes #10098

### CODE REVIEW

File: scene/2d/canvas_modulate.cpp [hunk 1]
```diff
@@ -70,7 +70,7 @@ void CanvasModulate::_bind_methods() {
 void CanvasModulate::set_color(const Color &p_color) {
 
 	color = p_color;
-	if (is_inside_tree()) {
+	if (is_inside_tree() && is_visible()) {
```
Comment (akien-mga) [inline scene/2d/canvas_modulate.cpp]: This should be `is_visible_in_tree()` to take into account the overall visibility given its parents visibility (`is_visible()` only checks that node's `visible` property).

And then no need for `is_inside_tree()` as it's implied by `is_visible_in_tree()`.

Comment (KoBeWi) [reply scene/2d/canvas_modulate.cpp]: Changed.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**